compare_mt.py

```diff
@@ -30,36 +31,26 @@ def print_score_report(ref, out1, out2,
     score_type: A string specifying the scoring type (bleu/length)
   """
   scorer = scorers.create_scorer_from_profile(score_type)
-  print(f'{scorer.name()}:')
   score1, str1 = scorer.score_corpus(ref,out1)
   score2, str2 = scorer.score_corpus(ref,out2)
-  if str1 is not None:
-    print(f' Sys1: {score1} ({str1})\n Sys2: {score2} ({str2})')
-  else:
-    print(f' Sys1: {score1}\n Sys2: {score2}')
+  stats = {'scorer_name': scorer.name(), 'score1': score1, 'str1': str1, 'score2': score2, 'str2': str2}
 
   if int(bootstrap) > 0:
-    print('Significance test. This may take a while.')
+    # print('Significance test. This may take a while.')
     wins, sys1_stats, sys2_stats = sign_utils.eval_with_paired_bootstrap(ref, out1, out2, score_type=score_type, num_samples=int(bootstrap))
+    stats = {**stats, 'wins': wins, 'sys1_stats': sys1_stats, 'sys2_stats': sys2_stats}
 
-    print('Win ratio: Sys1=%.3f, Sys2=%.3f, tie=%.3f' % (wins[0], wins[1], wins[2]))
-    if wins[0] > wins[1]:
-      print('(Sys1 is superior with p value p=%.3f)\n' % (1-wins[0]))
-    elif wins[1] > wins[0]:
-      print('(Sys2 is superior with p value p=%.3f)\n' % (1-wins[1]))
-
-    print('Sys1: mean=%.3f, median=%.3f, 95%% confidence interval=[%.3f, %.3f]' %
-            (sys1_stats['mean'], sys1_stats['median'], sys1_stats['lower_bound'], sys1_stats['upper_bound']))
-    print('Sys2: mean=%.3f, median=%.3f, 95%% confidence interval=[%.3f, %.3f]' %
-            (sys2_stats['mean'], sys2_stats['median'], sys2_stats['lower_bound'], sys2_stats['upper_bound']))
+  reporter = reporters.create_reporter_from_profile(profile='score', stats=stats)
+  reporter.print()
```

Comment:
Maybe change `print()` and `plot(...)` to `generate_report(fig_name=f'score-{score_type}-{bootstrap}')` without the final ".pdf" on the file name. The reason to avoid PDF is we might want to write out other file formats (e.g. PNG) to be added to a graphical HTML report.

In the case where we are writing out a graphical report we can pass along a tuple:
`report_files=(output_html_file, output_directory)` where `output_html_file` is an open handle to an HTML file, and `output_directory` is a path to the directory where we want to write the report. In the case where `report_files=None`, we don't write out any plots.

Reply:
Sure. I can change the functions as you suggest. Thanks for the suggestions!